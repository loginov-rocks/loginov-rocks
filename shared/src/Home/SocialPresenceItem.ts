export interface SocialPresenceItem {
  title: string;
  url: string;
  url2?: string;
}
