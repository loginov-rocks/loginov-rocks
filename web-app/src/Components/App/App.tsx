import { Link, Router } from '@reach/router';
import * as React from 'react';
import { addPrefetchExcludes, Root, Routes } from 'react-static';

import { DolphContainer } from 'containers/DolphContainer';

addPrefetchExcludes([
  'dolph',
]);

export const App: React.FunctionComponent = () => (
  <Root>
    {/* TODO: Remove */}
    <nav>
      <Link to="/">Home</Link>
      <Link to="/dolph">Dolph</Link>
    </nav>
    {/* TODO: Refactor */}
    <React.Suspense fallback={<em>Loading...</em>}>
      <Router>
        <DolphContainer path="dolph" />
        <Routes path="*" />
      </Router>
    </React.Suspense>
  </Root>
);

export default App;
