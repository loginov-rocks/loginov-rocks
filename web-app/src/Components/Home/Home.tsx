import { GitHubData } from '@loginov-rocks/loginov-rocks-shared';
import * as React from 'react';

import { SocialPresence } from 'Components/SocialPresence';
import { WorksList } from 'Components/WorksList';
import { WEB_APP_S3_GITHUB_FILE_KEY } from 'Constants';
import { HomeData } from 'Data/Interfaces/HomeData';
import { GitHubData as GetGitHubData } from 'Lib/GitHubData';

const getGitHubData = new GetGitHubData({ url: `/${WEB_APP_S3_GITHUB_FILE_KEY}` });

interface Props {
  data: HomeData;
}

export const Home: React.FunctionComponent<Props> = ({ data }) => {
  const [gitHubData, setGitHubData] = React.useState<GitHubData | null>(null);

  React.useEffect(() => {
    getGitHubData.get()
      .then((_gitHubData) => {
        setGitHubData(_gitHubData);
      });
  }, []);

  return (
    <>
      <h1>Hello!</h1>
      <p>
        I am
        {' '}
        <strong>Danila</strong>
        , nice to e-meet you!
      </p>
      <>
        <h2>Social Presence</h2>
        <SocialPresence items={data.socialPresenceItems} />
      </>
      <>
        <h2>Open Source</h2>
        <>
          <h3>Projects</h3>
          <WorksList gitHubRepos={gitHubData ? gitHubData.repos : []} works={data.openSource.projects} />
        </>
        <>
          <h3>Libraries</h3>
          <WorksList gitHubRepos={gitHubData ? gitHubData.repos : []} works={data.openSource.libraries} />
        </>
      </>
    </>
  );
};
