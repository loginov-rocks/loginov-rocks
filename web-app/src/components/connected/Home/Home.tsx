import * as React from 'react';

import { CmsConnectedProps } from 'cms/interfaces/CmsConnectedProps';
import { CmsEntry } from 'cms/interfaces/CmsEntry';

interface Props extends CmsConnectedProps {
  sections: CmsEntry[];
  title: string;
}

export const Home: React.FC<Props> = ({ render, sections, title }) => (
  <>
    <h1>{title}</h1>
    {render(sections)}
  </>
);
