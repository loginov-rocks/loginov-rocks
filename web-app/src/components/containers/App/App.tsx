import { Router } from '@reach/router';
import * as React from 'react';
import { addPrefetchExcludes, Root, Routes } from 'react-static';

import DolphPage from 'clientPages/dolph';
import { SplashView } from 'components/views/SplashView';

addPrefetchExcludes([
  'dolph',
]);

export const App: React.FC = () => (
  <Root>
    <React.Suspense fallback={<SplashView />}>
      <Router>
        <DolphPage path="dolph" />
        <Routes path="*" />
      </Router>
    </React.Suspense>
  </Root>
);
