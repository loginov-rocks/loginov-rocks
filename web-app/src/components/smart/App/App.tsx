import { Router } from '@reach/router';
import * as React from 'react';
import { addPrefetchExcludes, Root, Routes } from 'react-static';

import { DolphContainer } from 'components/dumb/DolphContainer';
import { Loading } from 'components/dumb/Loading';

addPrefetchExcludes([
  'dolph',
]);

export const App: React.FC = () => (
  <Root>
    <React.Suspense fallback={<Loading />}>
      <Router>
        <DolphContainer path="dolph" />
        <Routes path="*" />
      </Router>
    </React.Suspense>
  </Root>
);
