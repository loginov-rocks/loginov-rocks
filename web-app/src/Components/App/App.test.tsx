/* eslint-disable import/no-extraneous-dependencies */

import { shallow } from 'enzyme';
import * as React from 'react';

import * as homeData from 'Data/Home.json';

import { App } from './App';

it('matches snapshot', () => {
  const wrapper = shallow(<App homeData={homeData} />);

  expect(wrapper).toMatchSnapshot();
});
