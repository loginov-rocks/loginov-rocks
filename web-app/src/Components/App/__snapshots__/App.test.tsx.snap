// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<Root>
  <nav>
    <Link
      to="/"
    >
      Home
    </Link>
    <Link
      to="/dolph"
    >
      Dolph
    </Link>
  </nav>
  <Suspense
    fallback={
      <em>
        Loading...
      </em>
    }
  >
    <Router>
      <DolphContainer
        path="dolph"
      />
      <Routes
        path="*"
      />
    </Router>
  </Suspense>
</Root>
`;
